_posts/2018-11-30-release-1.7.0.md

```diff
@@ -0,0 +1,92 @@
+---
+layout: post
+title:  "Apache Flink 1.7.0 Release Announcement"
+date:   2018-11-30 12:00:00
+categories: news
+authors:
+- till:
+  name: "Till Rohrmann"
+  twitter: "stsffap"
+---
+
+The Apache Flink community is pleased to announce Apache Flink 1.7.0. 
+The latest release includes more than 420 resolved issues and some exciting additions to Flink that we describe in the following sections of this post. 
+Please check the [complete changelog](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12315522&version=12343585) for more details.
+
+
+Flink 1.7.0 is API-compatible with previous 1.x.y releases for APIs annotated with the `@Public` annotation.
+The release is available now and we encourage everyone to [download the release](http://flink.apache.org/downloads.html) and check out the updated [documentation](https://ci.apache.org/projects/flink/flink-docs-release-1.7/).
+Feedback through the Flink [mailing lists](http://flink.apache.org/community.html#mailing-lists) or [JIRA](https://issues.apache.org/jira/projects/FLINK/summary) is, as always, very much appreciated!
+
+You can find the binaries on the updated [Downloads page](http://flink.apache.org/downloads.html) on the Flink project site.
+
+{% toc %}
+
+## Flink 1.7.0 - Extending the reach of Stream Processing
+
+In Flink 1.7.0 we come closer to our goals of enabling fast data processing and building data-intensive applications for the Flink community in a seamless way. 
+Our latest release includes some exciting new features and improvements such as support for Scala 2.12, an exactly-once S3 file sink, the integration of complex event processing with streaming SQL and more features that we explain below.
+
+## New Features and Improvements
+
+* **Scala 2.12 Support in Apache Flink** ([FLINK-7811](https://issues.apache.org/jira/browse/FLINK-7811)):
+Apache Flink 1.7.0 is the first release which comes with full support for Scala 2.12. 
+This allows users to write Flink applications with a newer Scala version and to leverage the Scala 2.12 ecosystem.
+
+* **State Evolution** ([FLINK-9376](https://issues.apache.org/jira/browse/FLINK-9376)):
+In many cases, a long-running Flink application needs to evolve during its lifetime because of changing requirements. 
+Changing the user state without losing the current application progress in the form of its state is a crucial requirement for application evolution.
+
+  With Flink 1.7.0, the community added state evolution which allows you to flexibly adapt the user state of a long-running application while maintaining compatibility with previous savepoints. 
+  For example, with state evolution it is possible to add or remove columns to your schema in order to change which business features will be captured by your application after it has been deployed.
```

Comment:
I feel like that this paragraph should probably be rephrased a little, so that users have the right expectation for the new feature.

I suggest something along the lines of:
```
With Flink 1.7.0, the community added state evolution which allows you to flexibly adapt a long-running application's user states schema, while maintaining compatibility with previous savepoints. With state evolution it is possible to add or remove columns to your state schema in order to change which business features will be captured by your application after it has been deployed.

State schema evolution now works out-of-the-box when using Avro's generated classes as user state, meaning that the schema of the state can be evolved according to Avro's specifications. While Avro types are the only built-in type that supports schema evolution as of Flink 1.7, the community has continuous work to further extend support to other types in future Flink releases.
```

Overall, the general idea is to convey that:
- There is a framework available to request state schema evolution, if you're using custom serializers.
- For built-in types, currently only Avro is supported for schema evolution, while the community is working to further extend the support in future versions.

Reply:
Good point @tzulitai. I will update the paragraph.